src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -20,11 +20,47 @@ internal class GrpcHttpRequestData : HttpRequestData, IAsyncDisposable, IDisposa
         private HttpHeadersCollection? _headers;
         private Stream? _bodyStream;
         private bool _disposed;
+        private readonly Lazy<IReadOnlyCollection<IHttpCookie>> _cookies;
 
         public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             : base(functionContext)
         {
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
+            _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
+            {
+                if(Headers is null)
+                {
+                    return Array.Empty<IHttpCookie>();
+                }
+
+                var cookieString = Headers.FirstOrDefault(item => String.Equals(item.Key, "Cookie")).Value;
+
+                if (cookieString != null && cookieString.Any())
+                {
+
+                    return ToHttpCookies(cookieString.First().ToString());
```

Comment:
Is the `ToString()` call really needed ? `cookieString.First()` is returning a string.